ocaml/typing/ctype.ml

```diff
@@ -889,19 +889,20 @@ let update_scope_for tr_exn scope ty =
 *)
 
 let app_params_of_decl decl =
-  match decl.type_arity, decl.type_jkind with
+  match decl.type_arity, Jkind.get decl.type_jkind with
```

Comment:
It'd be nice to do it that way, but you point to a use site where this actually matters - `newgenvar` expects a full `Jkind.t` (with a history, so that the variable can have a jkind history), rather than a `Jkind.Jkind_desc.t`. I used `Jkind.get` throughout where one of the consumers wants to keep track of the history.